subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -70,7 +70,8 @@ public void add(final T operation) {
             workQueue.add(operation);
             pendingOperations++;
             workAvailable.signalAll();
-            if (workerCount == 0 || workerCount < workerLeases.getMaxWorkerCount()) {
+            if (workerCount == 0 || workerCount < workerLeases.getMaxWorkerCount() - 1) {
```

Comment:
Actually this didn't fix this problem when `--max-worker=1`. This might be the cause of #3844 @ghale WDYT?

Reply:
Actually, I think #3844 is because we now check the max workers parameter when the build starts and then pass the value around via a separate `ParallelismConfiguration` rather than passing it around via `StartParameter`.  (I'll comment on that issue)

Why do you think there is a problem when max-worker=1?  Is it because of that issue or because of other evidence?